database/database.go

```diff
@@ -79,11 +79,13 @@ var globalMutex Mutex = Mutex{0, 0, false}
 /* -------------------------------------------------------------------------- */
 
 // Connect establishes the connection to the PostgresSQL database and therefore
-// needs to be called before any other function of database.go
+// needs to be called before any other function of database.go.
 //
 // Notice: Connect doesn't initialize any tables or the cache, hence Initialize should be called
 // right afterwards.
-func Connect() error {
+//
+// Possible return states: StatusOK, StatusError
+func Connect() Status {
```

Comment:
this is un-go-like, the function should return an "optional" error value (it's nil when everything worked fine).
See `type SyntaxError struct {` code sample on https://blog.golang.org/error-handling-and-go, I recommend proceeding that way instead.

Reply:
we can use custom error types for detecting special cases

Reply:
this would work as in the code sample https://play.golang.org/p/vATgyXkWIt of https://medium.com/@sebdah/go-best-practices-error-handling-2d15e1f0c5ee when we use different error types for different cases. they can each yield special information of interest - a DatabaseError could contain a reference to the original database error and not include any extra text, for instance

Reply:
i addressed this in commit 4dca60a